ENIGMAsystem/SHELL/Universal_System/Extensions/Json/short_json_converter.cpp

```diff
@@ -0,0 +1,548 @@
+*
+*********************************************/
+
+std::string ShortJSONConverter::format_human_readable_error_message(std::string& message) { return ""; }
+
+/********************************************
+*
+*    Main system.
+*
+*********************************************/
+
+void ShortJSONConverter::push_level() {
+  levels_accumulators_.push("");
+  levels_states_.push(arrayOfObjectsState);  // Assume worst case which is "array of objects".
+
+  levels_entities_keys_indices_.push(0);
+
+  levels_boundaries_pointers_.push({virtual_pointer_, virtual_pointer_});
+
+  std::queue<std::pair<size_t, size_t>> level_entities_boundaries_pointers_;
+  levels_entities_boundaries_pointers_.push(level_entities_boundaries_pointers_);
+
+  std::queue<size_t> level_entities_offsets_;
+  levels_entities_offsets_.push(level_entities_offsets_);
+
+  levels_cumulative_offsets_.push(0);
+}  // push_level
+
+void ShortJSONConverter::pop_level() {
+  levels_accumulators_.pop();
+  levels_states_.pop();
+
+  /*
+    Pop "levels_entities_boundaries_pointers_" and "levels_entities_offsets_" current level, 
+    so that current level boundaries are added to previous level entities as well as offsets.
+
+    For example:
+
+                1       5   7       11
+                |       |   |       |
+              [ [ 1 , 4 ] , [ 1 , 4 ] , 4 ]  ---> this level entities boundaries should be {{1 , 5 } , { 7 , 11}}.
+  */
+  levels_entities_boundaries_pointers_.pop();
+  levels_entities_offsets_.pop();
+
+  /*
+    As long as it's an array, keep pushing because it maybe an "object" eventually.
+    
+    If it's an "object", then no need to continue pushing.
+  */
+  if (!(levels_entities_boundaries_pointers_.empty()) && !(levels_states_.empty())) {
+    if (levels_states_.top() != objectState)
+      levels_entities_boundaries_pointers_.top().push({levels_boundaries_pointers_.top()});
+  }
+
+  levels_boundaries_pointers_.pop();
+
+  levels_entities_keys_indices_.pop();
+}  // pop_level
+
+void ShortJSONConverter::accumulate_single_key() {
+  levels_accumulators_.top() += '\"';
+  levels_accumulators_.top() += std::to_string(levels_entities_keys_indices_.top());
+  levels_accumulators_.top() += '\"';
+  levels_accumulators_.top() += ':';
+}  // accumulate_single_key
+
+bool ShortJSONConverter::accumulate_value() {
+  size_t temp_pointer_{pointer_};
+
+  switch (data_.at(temp_pointer_)) {
+    case '\"': {
+      if (!accumulate_string_value()) return false;
+      break;
+    }
+    case '0':
+    case '1':
+    case '2':
+    case '3':
+    case '4':
+    case '5':
+    case '6':
+    case '7':
+    case '8':
+    case '9':
+    case '-': {
+      if (!accumulate_number_value()) return false;
+      break;
+    }
+    case 't':
+    case 'f': {
+      if (!accumulate_boolean_value()) return false;
+      break;
+    }
+    case 'n': {
+      if (!accumulate_null_value()) return false;
+      break;
+    }
+    default:
+      return false;
+  }
+
+  return true;
```

Comment:
Given that every branch of the `switch` has a return in it, this statement looks unreachable to me.

Reply:
actually, those return statements may not be invoked if the parsing is successful, break will be invoked instead.

Reply:
Ahh sorry, I didn't look carefully enough... somehow missed that